php/libraries/User.class.inc

```diff
@@ -66,19 +70,33 @@ class User extends UserPermissions
         $row = $DB->pselectRow($query, array('UID' => $username));
 
         // get user sites
-        $user_centerID_query =  $DB->pselect(
-            "SELECT CenterID FROM user_psc_rel upr
-                        WHERE upr.UserID= :UID",
-            array('UID' => $row['ID'])
-        );
+        $user_cid = array();
+        if (!isset($row['ID'])) {
+            error_log(
+                "Could not get user center ID(s) because'
+                . ' user DB query value 'ID' is not set."
+            );
```

Comment:
I don't think this is possible. The users.ID column isn't nullable... which also means this should probably be an exception, not an error if it does need to stay for some reason.